neural_compressor/adaptor/torch_utils/smooth_quant.py

```diff
@@ -902,24 +903,22 @@ def update_sq_scale(ipex_config_path, smoothquant_scale_info):
         for module_name, v in ipex_config.items():
             if 'q_op_infos' in v and v['q_op_infos']:
                 for op_num, v1 in v['q_op_infos'].items():
-                    if 'weight_tensor_infos' in v1 and v1['weight_tensor_infos']:
-                        op_name = v1['fqn']
-                        if op_name in smoothquant_scale_info:
-                            input_scale_for_mul = \
-                                    smoothquant_scale_info[op_name]['input_scale_for_mul'].tolist()
-                            input_scale_after_mul = \
-                                    smoothquant_scale_info[op_name]['input_scale_after_mul'].tolist()
-                            input_zero_point_after_mul = \
-                                    smoothquant_scale_info[op_name]['input_zero_point_after_mul'].tolist()
-                            weight_scale_for_mul = \
-                                    (1 / smoothquant_scale_info[op_name]['input_scale_for_mul']).tolist()
-                            weight_scale_after_mul = \
-                                    smoothquant_scale_info[op_name]['weight_scale_after_mul'].tolist()
-                            v1['input_tensor_infos'][0]['smooth_quant_scaling_factor'] = input_scale_for_mul
-                            v1['input_tensor_infos'][0]['scale'] = input_scale_after_mul
-                            v1['input_tensor_infos'][0]['zero_point'] = input_zero_point_after_mul
-                            v1['weight_tensor_infos'][0]['smooth_quant_scaling_factor'] = weight_scale_for_mul
-                            v1['weight_tensor_infos'][0]['scale'] = weight_scale_after_mul
+                    # update alpha data instead of updating weight scale
+                    op_name = v1['fqn'] # fqn always exists even it's empty.
+                    if op_name in smoothquant_scale_info:
+                        # observers were overridden by the fallback step, setting it back.
```

Comment:
is it better to move this function to ipex backend?

Reply:
I see your point, I will move it to utils.py next time.